### PR TITLE
Consolidate position tracking into Hex class

### PR DESCRIPTION
**Pull request by CyberiaResurrection**

### CODE REVIEW

File: PyRoute/Star.py [hunk 1]
```diff
@@ -97,17 +100,14 @@ class Star(object):
...
     def __init__(self):
         self.logger = logging.getLogger('PyRoute.Star')
         self._hash = None
         self._key = None
         self.component = None
         self.index = None
-        self.x = None
-        self.y = None
-        self.z = None
         self.q = None
```
Comment (tjoneslo) [inline PyRoute/Star.py:111]: You should be able remove the q/r coordinates from here as well. Everything should be referencing the Hex class.

File: PyRoute/Star.py [hunk 2]
```diff
@@ -249,14 +251,13 @@ def parse_line_into_star(line, sector, pop_code, ru_calc):
...
-        if (data[5]):
+        if data[5]:
             imp = int(data[5][1:-1].strip())
             star.calculate_importance()
             if imp != star.importance:
                 star.logger.warning(
-                    '{}-{} Calculated importance {} does not match generated importance {}'.format(star, star.baseCode,
-                                                                                                    star.importance,
-                                                                                                    imp))
+                    '{}-{} Calculated importance {} does not match generated importance {}'.
```
Comment (tjoneslo) [inline PyRoute/Star.py:259]: Please use the python format strings whenever possible. But only if you end up changing them. You can remove the call to format(...) as well. 
```suggestion
                    f'{star}-{star.baseCode} Calculated importance {star.importance} does not match generated importance {imp}'
```

File: PyRoute/Position/Hex.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import functools
+
+
+class Hex(object):
+
+    position = None  # Hex location
+    row = 0  # Location in the sector.
+    col = 0
+    dx = 0  # location in the whole space, row/column coordinate
+    dy = 0
```
Comment (tjoneslo) [inline PyRoute/Position/Hex.py:15]: As I recall, but still need to dig into this, but we may not need the dx/dy value. In the original version of this, the x/y were used to calculate distances and the q/r values were used by the border drawing. But if you're using the q/r for distance calculations, the dx/dy are unused and you should be able to remove them. 

See also updating the Star class to move the q/r uses entirely here.

**Review by tjoneslo — CHANGES_REQUESTED** [review]
Another round of cleanup.

File: PyRoute/Star.py (hunk 2 above)
Comment (CyberiaResurrection) [reply PyRoute/Star.py:259]: I inherited this one from you, so I didn't substantively touch it.  Maybe do another PR that just straightens all of this string stuff out?

File: PyRoute/Position/Hex.py (hunk 3 above)
Comment (CyberiaResurrection) [reply PyRoute/Position/Hex.py:15]: I'll leave it in unless/until you decide it needs to go bouncing down the road.

File: PyRoute/Star.py (hunk 1 above)
Comment (CyberiaResurrection) [reply PyRoute/Star.py]: That's all ripped out, replaced by properties wrapping the corresponding values in the Hex object

**Review by tjoneslo — APPROVED** [review]

**Pull request merged**